### PR TITLE
[SMALLFIX] Consistently use basenames for UfsStatus names

### PR DESCRIPTION
**Pull request by aaudiber**
For getStatus, we always set the `UfsStatus` name to the full path, but never seem to use it anywhere.
For listStatus, we always set the `UfsStatus` name to the basename, and do sometimes use it.
It would be nice if we could be consistent with what type of path we store in `UfsStatus`. Given that we currently rely on listStatus using basenames, I propose using basename consistently.

### CODE REVIEW

**Comment by aaudiber** [comment]
Jenkins, test this please

File: core/common/src/main/java/alluxio/util/io/PathUtils.java [hunk 1]
```diff
@@ -37,6 +37,41 @@
...
+   *
+   * Examples:
+   *
+   * <pre>
+   *   basename("/") = "/"
+   *   basename("/abc") = "abc"
+   *   basename("///") = "/"
+   *   basename("abc/" = "abc"
+   *   basename("") = ""
+   *   basename("/a.txt") = "a.txt"
```
Comment (gpang) [inline core/common/src/main/java/alluxio/util/io/PathUtils.java]: can you provide nested examples, like /a/b/c?

File: core/common/src/main/java/alluxio/util/io/PathUtils.java [hunk 2]
```diff
@@ -37,6 +37,41 @@
...
+  /**
+   * Returns the basename of the path, consistent with the Unix basename command.
+   *
+   * Examples:
+   *
+   * <pre>
+   *   basename("/") = "/"
+   *   basename("/abc") = "abc"
+   *   basename("///") = "/"
+   *   basename("abc/" = "abc"
```
Comment (gpang) [inline core/common/src/main/java/alluxio/util/io/PathUtils.java]: missing closing `)`

Comment (aaudiber) [reply core/common/src/main/java/alluxio/util/io/PathUtils.java]: done

File: core/common/src/main/java/alluxio/util/io/PathUtils.java (hunk 1 above)
Comment (aaudiber) [reply core/common/src/main/java/alluxio/util/io/PathUtils.java]: done

**Comment by aaudiber** [comment]
We actually do both, logging a warning if the caller passed a non-basename. I preferred this approach so that users aren't surprised when their names are base-ified.

**Pull request merged**